addons/crm/tests/test_crm_lead_lost.py

```diff
@@ -121,12 +119,15 @@ def test_lead_lost_crm_rights(self):
                 'name': 'Test Reason'
             })
 
-        lost_wizard = self.env['crm.lead.lost'].with_context({
-            'active_ids': lead.ids
-        }).create({
+        lost_wizard = self.env['crm.lead.lost'].create({
+            'lead_ids': lead.ids,
             'lost_reason_id': lost_reason.id
         })
 
         # nice try little salesman, you cannot invoke a wizard to update other people leads
         with self.assertRaises(AccessError):
+            # because assertRaises clears the cursor, we need to load the leads back in
+            # we use sudo to prevent `_apply_ir_rules` from adding query conditions to prevent
+            # loading 'invalid' leads, as we want them to fail here
+            lost_wizard.sudo().leads
             lost_wizard.action_lost_reason_apply()
```

Comment:
2 alternatives:

- Do the assertRaises manually with a `try: catch:` where we `assertTrue(False)` in the `try` block, which is not obvious but isn't affected by cache nonsense

- Clear the cache manually first, then check that the leads were not marked as lost. But this is reliant on caching/fetching behaviour, so not great